---
title: "Class15"
author: "Ayse"
date: '2022-03-09'
output:
  pdf_document: default
  html_document: default
---

    Q1. With the help of the R “addin” package datapasta assign the CDC pertussis case number data to a data frame called cdc and use ggplot to make a plot of cases numbers over time.

```{r}
#install.packages("datapasta")
library(datapasta)
cdc <- data.frame(
                          Year = c(1922L,
                                   1923L,1924L,1925L,1926L,1927L,1928L,
                                   1929L,1930L,1931L,1932L,1933L,1934L,1935L,
                                   1936L,1937L,1938L,1939L,1940L,1941L,
                                   1942L,1943L,1944L,1945L,1946L,1947L,1948L,
                                   1949L,1950L,1951L,1952L,1953L,1954L,
                                   1955L,1956L,1957L,1958L,1959L,1960L,
                                   1961L,1962L,1963L,1964L,1965L,1966L,1967L,
                                   1968L,1969L,1970L,1971L,1972L,1973L,
                                   1974L,1975L,1976L,1977L,1978L,1979L,1980L,
                                   1981L,1982L,1983L,1984L,1985L,1986L,
                                   1987L,1988L,1989L,1990L,1991L,1992L,1993L,
                                   1994L,1995L,1996L,1997L,1998L,1999L,
                                   2000L,2001L,2002L,2003L,2004L,2005L,
                                   2006L,2007L,2008L,2009L,2010L,2011L,2012L,
                                   2013L,2014L,2015L,2016L,2017L,2018L,
                                   2019L),
  No..Reported.Pertussis.Cases = c(107473,
                                   164191,165418,152003,202210,181411,
                                   161799,197371,166914,172559,215343,179135,
                                   265269,180518,147237,214652,227319,103188,
                                   183866,222202,191383,191890,109873,
                                   133792,109860,156517,74715,69479,120718,
                                   68687,45030,37129,60886,62786,31732,28295,
                                   32148,40005,14809,11468,17749,17135,
                                   13005,6799,7717,9718,4810,3285,4249,
                                   3036,3287,1759,2402,1738,1010,2177,2063,
                                   1623,1730,1248,1895,2463,2276,3589,
                                   4195,2823,3450,4157,4570,2719,4083,6586,
                                   4617,5137,7796,6564,7405,7298,7867,
                                   7580,9771,11647,25827,25616,15632,10454,
                                   13278,16858,27550,18719,48277,28639,
                                   32971,20762,17972,18975,15609,18617)
)
library(ggplot2)
ggplot(cdc) +
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_line() +
  labs(title="Pertussis Cases by Year")
```

    Q2. Using the ggplot geom_vline() function add lines to your previous plot for the 1946 introduction of the wP vaccine and the 1996 switch to aP vaccine (see example in the hint below). What do you notice?

```{r}
ggplot(cdc) +
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_vline(xintercept = 1946, linetype = "longdash") +
  geom_vline(xintercept = 1996, linetype = "longdash") +
  geom_line() +
  labs(title="Pertussis Cases by Year")
```


    Q3. Describe what happened after the introduction of the aP vaccine? Do you have a possible explanation for the observed trend?
At the first introduction of the aP vaccine, there seemed to be a continuation of the minimal cases. However, slight increases in the number of cases can be seen after 2000. This may be due to a drop in vaccination rates, loss of immunity over the years for patients who were vaccinated before 2000, or more sensitive tests to detect pertussis being developed after the year 2000.


```{r}
# Allows us to read, write and process JSON data
library(jsonlite)
subject <- read_json("https://www.cmi-pb.org/api/subject", simplifyVector = TRUE) 
head(subject, 3)
```


    Q4. How many aP and wP infancy vaccinated subjects are in the dataset?
47 for aP and 49 for wP.

```{r}
table(subject$infancy_vac)
```
    Q5. How many Male and Female subjects/patients are in the dataset?
    66 female and 30 male.
```{r}
table(subject$biological_sex)
```
    Q6. What is the breakdown of race and biological sex (e.g. number of Asian females, White males etc…)?
    The breakdown is as shown in the table below.
```{r}
table(subject$race, subject$biological_sex)
```

Q7. Using this approach determine (i) the average age of wP individuals, (ii) the average age of aP individuals; and (iii) are they significantly different?
    i. 35
    ii. 25
    iii. yes
```{r}
library(lubridate)
#add column age showing the age of each subject
subject$age <- time_length(today() - ymd(subject$year_of_birth), "years")

library(dplyr)
#filter for individuals that were vaccinated with aP
ap <- subject %>% filter(infancy_vac == "aP")

mean_ap <- mean(ap$age)
mean_ap

#filter for individuals that were vaccinated with wP
wp <- subject %>% filter(infancy_vac == "wP")

mean_wp <- mean(wp$age)
mean_wp
```
    Q8. Determine the age of all individuals at time of boost?
    Ages are listed below. Average age of boost is around 25.
    
```{r}
subject$boostage <- time_length(ymd(subject$date_of_boost) - ymd(subject$year_of_birth), "years")
subject$boostage
mean(subject$boostage)
```
    
    Q9. With the help of a faceted boxplot or histogram (see below), do you think these two groups are significantly different?
 yes, there is no overlap
```{r}
ggplot(subject) +
  aes(time_length(age, "year"),
      fill=as.factor(infancy_vac)) +
  geom_histogram(show.legend=FALSE) +
  facet_wrap(vars(infancy_vac), nrow=2) 
```

```{r}
specimen <- read_json("https://www.cmi-pb.org/api/specimen", simplifyVector = TRUE) 
titer <- read_json("https://www.cmi-pb.org/api/ab_titer", simplifyVector = TRUE) 
```


    Q9. Complete the code to join specimen and subject tables to make a new merged data frame containing all specimen records along with their associated subject details:
  
```{r}
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```
    Q10. Now using the same procedure join meta with titer data so we can further analyze this data in terms of time of visit aP/wP, male/female etc.
    
```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
```

    Q11. How many specimens (i.e. entries in abdata) do we have for each isotype?
Shown in the table below.
```{r}
table(abdata$isotype)
```

    Q12. What do you notice about the number of visit 8 specimens compared to other visits?
  there are much fewer visit 8 specimens compared to the other visits.
```{r}
table(abdata$visit)
```
    
```{r}
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

    Q13. Complete the following code to make a summary boxplot of Ab titer levels for all antigens:

```{r}
ggplot(ig1) +
  aes(ab_titer, antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2)
```
    Q14. What antigens show differences in the level of IgG1 antibody titers recognizing them over time? Why these and not others?
FIM2/3, FHA, PRN. These must have been low due to waning of immunity and upon boosting, increased.

```{r}
ggplot(ig1) +
  aes(ab_titer, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2) +
  theme(axis.text.x = element_text(angle = 45, hjust=1))
```

    Q15. Filter to pull out only two specific antigens for analysis and create a boxplot for each. You can chose any you like. Below I picked a “control” antigen (“Measles”, that is not in our vaccines) and a clear antigen of interest (“FIM2/3”, extra-cellular fimbriae proteins from B. pertussis that participate in substrate attachment).
    
```{r}
filter(ig1, antigen=="Measles") %>%
  ggplot() +
  aes(ab_titer, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
    
```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(ab_titer, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```


    Q16. What do you notice about these two antigens time courses and the FIM2/3 data in particular?
    
FIM2/3 titer levels are much higher than Measles and peaks at visit 5 for both wP and aP vaccinated subjects.
    
    Q17. Do you see any clear difference in aP vs. wP responses?

No.

```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE) 

ssrna <- inner_join(rna, meta)
```


    Q18. Make a plot of the time course of gene expression for IGHG1 gene (i.e. a plot of visit vs. tpm).

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```

    Q19.: What do you notice about the expression of this gene (i.e. when is it at it’s maximum level)?

The expression of this gene is at its maximal level at visit 4.

    Q20. Does this pattern in time match the trend of antibody titer data? If not, why not?
No, because gene expression decreases after it peaks, whereas for the antibody titers, they stayed stable and did not decrease after they peaked.

```{r}
ggplot(ssrna) +
  aes(tpm, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit))
```

```{r}
ssrna %>%  
  filter(visit==4) %>% 
  ggplot() +
    aes(tpm, col=infancy_vac) + geom_density() + 
    geom_rug()
```

